# Impact of Amazon-Occupied Buildings on Seattle's GHGE

## Todo:

- List of Amazon-occupied properties
    - Anything with Amazon in the name
    - Anything from [this article](https://www.geekwire.com/amazon-office-buildings/)
    - Data center at the Westin (used by Amazon)
    - Whole Foods stores, Amazon Go stores

- Add a category for office building/ata center/Whole Foods/Amazon Go

- Total GHGE, compare with other categories (office buildings), other owners

- Site/source GHGE

- Compare electricity and gas use

- Compare GHG Intensity to categories/owners

- Compare Whole Foods--is it worse, better, etc.?


## If this isn't fruitful:

Find a new villain: 

- What categories are the biggest polluters?
- What owners are the biggest polluters (pull from landlords.csv)
- What buildings/owners/types are secretly polluting a lot or in a strange way?

In [2]:
import numpy as np
import pandas as pd
import pandas_gbq

In [3]:
query = """
    SELECT *
    FROM `GHGE_buildings_data.building-benchmark`;
"""

all_buildings_dataframe = pandas_gbq.read_gbq(query, project_id='seattle-377109')

Downloading: 100%|███████████████████████████████████████████████████████████████████████████████████|


In [4]:
# df of landlords' overall energy use, uses updated owners
aggregate_landlords_df = pd.read_csv('../worst_offenders/landlords_with_total_energy_use_2_16_23.csv')

In [5]:
aggregate_landlords_df.head()

Unnamed: 0                             Owner  BuildingsOwned  \
0           0                   CITY OF SEATTLE             175   
1           1          UNIVERSITY OF WASHINGTON              43   
2           2               STATE OF WASHINGTON              17   
3           3                       KING COUNTY              16   
4           4  OVERLOOK MAGNOLIA APARTMENTS LLC              11   

   TotalSquareFootage  TotalGHGEmissions__metric_tons_  \
0            17038324                            16357   
1            18445870                            39712   
2             2768746                             3614   
3             4355059                            22535   
4              388815                               52   

   TotalElectricity_kBtu_  TotalSteamUse_kBtu  TotalNaturalGas_kBtu  \
0               501287744             6826192             247129120   
1               679730875           366595771             106707401   
2                88426130                   0              59208493   
3               255701627            16515088             372904431   
4                 9911319                   0                     0   

   TotalOtherFuelUse_kBtu  AverageGHGEmissionsIntensity  \
0                       0                      1.172674   
1                       0                      4.756098   
2                       0                      2.112500   
3                       0                      1.750000   
4                       0                      0.127273   

   AverageSiteEUI_kBtu_sf  AverageSourceEUI_kBtu_sf  AverageENERGYSTARScore  
0               49.970175                108.376608               72.747899  
1              147.265000                321.932500               61.818182  
2               71.162500                134.625000               55.375000  
3               66.943750                137.293750               68.083333  
4               25.418182                 71.145455               74.545455

In [6]:
# all buildings with consolidated owners

updated_owners_df = pd.read_csv('../worst_offenders/updated_owners_2_15_23.csv')

updated_owners_df.head()

Unnamed: 0  OSEBuildingID TaxParcelIdentificationNumber  DataYear  \
0           0          50148                    7733600135      2020   
1           1          50150                    1991200090      2020   
2           2          50160                     659000775      2020   
3           3          50166                    4083306985      2020   
4           4          50192                     660001605      2020   

     BuildingType             BuildingName                  Owner  \
0  NonResidential  RESIDENCE INN SEATTLE U           TMUD GSL LLC   
1  NonResidential              HYATT HOUSE         HH SEATTLE LLC   
2  NonResidential  AMAZON DOPPLER BUILDING  ACORN DEVELOPMENT LLC   
3  NonResidential     TABLEAU (NORTH EDGE)       BRE-BMR 34TH LLC   
4  NonResidential      MIDTOWN 21 (AMAZON)          MIDTOWN21 LLC   

   CouncilDistrictCode Neighborhood Units  ...  \
0                  4.0    NORTHEAST   NaN  ...   
1                  1.0   LAKE UNION   NaN  ...   
2                  7.0     DOWNTOWN   NaN  ...   
3                  1.0   LAKE UNION   NaN  ...   
4                  1.0     DOWNTOWN   NaN  ...   

   SecondLargestPropertyUseTypeGFA ThirdLargestPropertyUseType  \
0                          54695.0                         NaN   
1                          28069.0                         NaN   
2                         448625.0           Convention Center   
3                         117684.0                  Restaurant   
4                         110813.0                         NaN   

  ThirdLargestPropertyUseTypeGFA Outlier  ComplianceIssue  ComplianceStatus  \
0                            NaN     NaN         No Issue         Compliant   
1                            NaN     NaN         No Issue         Compliant   
2                        59672.0     NaN         No Issue         Compliant   
3                         2839.0     NaN         No Issue         Compliant   
4                            NaN     NaN         No Issue         Compliant   

   Comments  DefaultData  LegislationPropertyType Dept  
0       NaN          NaN                    Hotel  NaN  
1       NaN          NaN                    Hotel  NaN  
2       NaN          NaN                   Office  NaN  
3       NaN          NaN                   Office  NaN  
4       NaN          NaN                   Office  NaN  

[5 rows x 56 columns]

## List of Amazon-occupied properties

    - Anything with Amazon in the name
    - Anything from [this article](https://www.geekwire.com/amazon-office-buildings/)
    - Data center at the Westin (used by Amazon)
    - Whole Foods stores, Amazon Go stores

### Buildings with Amazon in the name

In [7]:
amazon_ose_ids = []

buildings_with_amazon_in_name = all_buildings_dataframe[all_buildings_dataframe['BuildingName'].str.contains('amazon', case=False)]

In [8]:
buildings_with_amazon_in_name.head()

OSEBuildingID TaxParcelIdentificationNumber  DataYear    BuildingType  \
2             50160                     659000775      2020  NonResidential   
4             50192                     660001605      2020  NonResidential   
1497          49985                    1983200075      2020  NonResidential   
1499          50037                    1991201265      2020  NonResidential   
1500          50045                    1983200035      2020  NonResidential   

                              BuildingName  \
2                  AMAZON DOPPLER BUILDING   
4                      MIDTOWN 21 (AMAZON)   
1497  AMAZON PHASE VI - NESSIE AND BIGFOOT   
1499                  APOLLO BLDG (AMAZON)   
1500                         AMAZON BRAZIL   

                                       Owner  CouncilDistrictCode  \
2                      ACORN DEVELOPMENT LLC                    7   
4                              MIDTOWN21 LLC                    1   
1497       UNION INVESTMENT REAL ESTATE GMBH                    1   
1499                     USQ PHASE EIGHT LLC                    1   
1500  RF SIDNEYSEA TIC LLC +BRAZIL 7STAR LLC                    1   

     Neighborhood Units  YearBuilt  ... SecondLargestPropertyUseType  \
2        DOWNTOWN  None       2016  ...                      Parking   
4        DOWNTOWN  None       2016  ...                      Parking   
1497   LAKE UNION  None       2014  ...                      Parking   
1499   LAKE UNION  None       2014  ...                      Parking   
1500   LAKE UNION  None       2015  ...                      Parking   

     SecondLargestPropertyUseTypeGFA ThirdLargestPropertyUseType  \
2                             448625           Convention Center   
4                             110813                        None   
1497                          169195                        None   
1499                          136594                Food Service   
1500                          146810                  Restaurant   

      ThirdLargestPropertyUseTypeGFA  Outlier  ComplianceIssue  \
2                              59672     None         No Issue   
4                               <NA>     None         No Issue   
1497                            <NA>     None         No Issue   
1499                            5699     None         No Issue   
1500                            5416     None         No Issue   

     ComplianceStatus Comments DefaultData  LegislationPropertyType  
2           Compliant     None        None                   Office  
4           Compliant     None        None                   Office  
1497        Compliant     None        None                   Office  
1499        Compliant     None        None                   Office  
1500        Compliant     None        None                   Office  

[5 rows x 54 columns]

In [9]:
amazon_ose_ids += list(buildings_with_amazon_in_name['OSEBuildingID'])

amazon_ose_ids

[50160,
 50192,
 49985,
 50037,
 50045,
 50241,
 50270,
 50528,
 50294,
 50656,
 19776,
 390]

### Buildings from GeekWire Article

https://www.geekwire.com/amazon-office-buildings/

This website is a WP site with an embedded Google Map. All normal ways of scraping Google Maps data appear not to work on WP sites, with no obvious substitutes (API responses, HTML, etc. all draw blanks). So let's copy-pasta the text and go with that.

In [10]:
amazon_headers = '''Office Name	Sq Ft	Address	Leased/Owned	News Coverage	Announced'''.split('	')
amazon_buildings_text = '''
	1260 Mercer	106,000	
1260 Mercer St.
Seattle, WA 98109
Lease	  Read More	2011
	1918 Eighth	460,000	
1918 Eighth Ave.
Seattle, WA 98101
Lease	  Read More	2011
	2201 Westlake	180,000	
2201 Westlake Ave.
Seattle, WA 98121
Lease	  Read More	2010
	300 Pine (Macy’s)	470,000	
300 Pine St.
Seattle, WA 98101
Lease	  Read More	2017
	425 Centre	354,000	
425 106th Ave. N.E.
Bellevue, WA 98004
Lease	  Read More	2016
	428 Westlake	81,000	
428 Westlake Ave. N.
Seattle, WA 98109
Lease	  Read More	2016
	4th Block	850,000	
2200 Seventh Ave.
Seattle, WA 98121
Own	  Read More	2014
	5th Block	405,000	
2205 Seventh Ave.
Seattle, WA 98121
Own	  Read More	2016
	9th & Thomas	156,000	
234 Ninth Ave. N.
Seattle, WA 98109
Lease	  Read More	2017
	Amazon Alexandria	200,000	
1800 Ninth Ave. N.
Seattle, WA 98101
Lease	  Read More	2013
	Amazon Coral	100,000	
2301 Fifth Ave.
Seattle, WA 98121
Lease	  Read More	2014
	Amazon Delight	19,000	
507 Westlake Ave. N.
Seattle, WA 98109
Lease	  Read More	2014
	Amazon Galaxy	196,000	
635 Elliott Ave. W.
Seattle, WA 98119
Lease	  Read More	2014
	Amazon Houston	17,000	
224 Westlake Ave. N.
Seattle, WA 98109
Lease		2013
	Amazon Kumo	251,000	
1915 Terry Ave.
Seattle, WA 98101
Lease	  Read More	2014
	Amazon Phase VI (Nessie)	392,000	
500 Ninth Ave. N.
Seattle, WA 98109
Lease	  Read More	2012
	Amazon Phase VII	313,000	
400 Ninth Ave. N.
Seattle, WA 98109
Lease	  Read More	2013
	Amazon Phase VIII	317,000	
325 Ninth Ave. N.
Seattle, WA 98109
Lease	  Read More	2013
	Amazon Roxanne	110,000	
202 Westlake Ave. N.
Seattle, WA 98109
Lease	  Read More	2012
	Amazon Stackhouse	26,000	
477 Yale Ave. N.
Seattle, WA 98109
Lease	  Read More	2014
	Blanchard Plaza	255,000	
2201 Sixth Ave.
Seattle, WA 98121
Lease	  Read More	2014
	Block 20	1.1 million	
2100 Seventh Ave.
Seattle, WA 98121
Own		2012
	Day One	1.1 million	
2121 Seventh Ave.
Seattle, WA 98121
Own	  Read More	2012
Amazon Doppler	Doppler	1.1 million	
2021 Seventh Ave.
Seattle, WA 98121
Own	  Read More	2012
	Former Pemco HQ	168,300	
301 Eastlake Ave. E.
Seattle, WA 98109
Lease	  Read More	2018
	Metropolitan Park North	140,000	
1220 Howell St.
Seattle, WA 98101
Lease	  Read More	2013
	Midtown21	365,000	
1007 Stewart St.
Seattle, WA 98101
Lease	  Read More	2016
	Original South Lake Union Campus (11 buildings)	1.7 million	
410 Terry Ave. N.
Seattle, WA 98109
Own	  Read More	2007
	Rainier Square redevelopment	710,000	
1301 5th Ave
Seattle, WA 98101
Lease	  Read More	2017
	Tilt49	306,000	
1812 Boren Ave.
Seattle, WA 98101
Lease	  Read More	2017
	Troy Block	817,000	
307 Fairview Ave. N.
Seattle, WA 98109
Lease	  Read More	2015
	Urban Union	287000	
501 Fairview Ave. N.
Seattle, WA 98109
Lease	  Read More	2016
	West 8th	250,000	
2001 Eighth Ave.
Seattle, WA 98121
Lease	  Read More	2011
	Westlake Terry	275,000	
321 Terry Ave. N.
Seattle, WA 98109
Lease	  Read More	2016'''

In [11]:
amazon_building_indiv = amazon_buildings_text.split('	') 
# this puts the announcement year in the next row, 
# but we don't actually care for this purpose

In [12]:
amazon_building_indiv

['\n',
 '1260 Mercer',
 '106,000',
 '\n1260 Mercer St.\nSeattle, WA 98109\nLease',
 '  Read More',
 '2011\n',
 '1918 Eighth',
 '460,000',
 '\n1918 Eighth Ave.\nSeattle, WA 98101\nLease',
 '  Read More',
 '2011\n',
 '2201 Westlake',
 '180,000',
 '\n2201 Westlake Ave.\nSeattle, WA 98121\nLease',
 '  Read More',
 '2010\n',
 '300 Pine (Macy’s)',
 '470,000',
 '\n300 Pine St.\nSeattle, WA 98101\nLease',
 '  Read More',
 '2017\n',
 '425 Centre',
 '354,000',
 '\n425 106th Ave. N.E.\nBellevue, WA 98004\nLease',
 '  Read More',
 '2016\n',
 '428 Westlake',
 '81,000',
 '\n428 Westlake Ave. N.\nSeattle, WA 98109\nLease',
 '  Read More',
 '2016\n',
 '4th Block',
 '850,000',
 '\n2200 Seventh Ave.\nSeattle, WA 98121\nOwn',
 '  Read More',
 '2014\n',
 '5th Block',
 '405,000',
 '\n2205 Seventh Ave.\nSeattle, WA 98121\nOwn',
 '  Read More',
 '2016\n',
 '9th & Thomas',
 '156,000',
 '\n234 Ninth Ave. N.\nSeattle, WA 98109\nLease',
 '  Read More',
 '2017\n',
 'Amazon Alexandria',
 '200,000',
 '\n1800 Ninth 

In [13]:
amazon_building_names = amazon_building_indiv[1:][::5]

amazon_building_names

['1260 Mercer',
 '1918 Eighth',
 '2201 Westlake',
 '300 Pine (Macy’s)',
 '425 Centre',
 '428 Westlake',
 '4th Block',
 '5th Block',
 '9th & Thomas',
 'Amazon Alexandria',
 'Amazon Coral',
 'Amazon Delight',
 'Amazon Galaxy',
 'Amazon Houston',
 'Amazon Kumo',
 'Amazon Phase VI (Nessie)',
 'Amazon Phase VII',
 'Amazon Phase VIII',
 'Amazon Roxanne',
 'Amazon Stackhouse',
 'Blanchard Plaza',
 'Block 20',
 'Day One',
 'Doppler',
 'Former Pemco HQ',
 'Metropolitan Park North',
 'Midtown21',
 'Original South Lake Union Campus (11 buildings)',
 'Rainier Square redevelopment',
 'Tilt49',
 'Troy Block',
 'Urban Union',
 'West 8th',
 'Westlake Terry']

In [14]:
from fuzzywuzzy import fuzz
from fuzzywuzzy import process

/usr/local/lib/python3.9/site-packages/fuzzywuzzy/fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning
  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')


In [15]:
len(amazon_ose_ids)

12

In [16]:
potential_matches = []

for building in amazon_building_names:
    df_match = list(all_buildings_dataframe[all_buildings_dataframe['BuildingName'].str.contains(building, case=False)]['OSEBuildingID'])
    if df_match:
        amazon_ose_ids += df_match
    else:
        potential_matches.append([building, process.extract(building, all_buildings_dataframe['BuildingName'], limit=10)])

/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_99492/1596756794.py:4: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  df_match = list(all_buildings_dataframe[all_buildings_dataframe['BuildingName'].str.contains(building, case=False)]['OSEBuildingID'])


In [17]:
potential_matches

[['300 Pine (Macy’s)',
  [('PINE ST. CONDO-ELLIOTT GRAND HYATT SEATTLE', 86, 70),
   ("SEATTLE CHILDREN'S HOSPITAL  (JACK R MCDONALD BUILDING)", 86, 80),
   ('5TH & PINE', 86, 82),
   ("SEATTLE CHILDREN'S HOSPITAL RESEARCH BUILDING #2", 86, 132),
   ('TWO PINE', 86, 134),
   ("FOURTH & UNION BUILDING/MEN'S WAREHOUSE", 86, 148),
   ('1ST & PINE', 86, 271),
   ('UNIVERSITY VILLAGE EAST BLDG S', 86, 950),
   ("BARTELL'S AND APARTMENTS", 86, 999),
   ("SEATTLE CHILDREN'S HOSPITAL  (MAIN CAMPUS)", 86, 1117)]],
 ['425 Centre',
  [('LEGACY CENTRE & ARBOR PLACE CONDO', 86, 2445),
   ('4245 ROOSEVELT - UW MEDICAL CENTER', 77, 959),
   ('US BANK CENTRE', 71, 105),
   ('CENTRE COURT', 71, 1650),
   ('MERIDIAN CENTER WEST', 63, 157),
   ('MERIDIAN CENTER EAST', 63, 158),
   ('UNIVERSITY VILLAGE SHOPPING CENTER WEST BLDG', 63, 937),
   ('PORT OF SEATTLE - WORLD TRADE CENTER-WEST', 63, 2390),
   ('WORLD TRADE CENTER-EAST', 63, 2391),
   ('NORTHGATE MALL SHOPPING CENTER RETAIL STORES  - ACTIVE/PRIMAR

In [18]:
# 4th block and 5th block aren't in our dataset in any form, probably because dataset is from 2020
# Ditto Amazon Alexandria, Amazon Delight
# Amazon Unicorn building is owned by the city! Any more like that? Or has it been sold?

matched_names = [
    'AMAZON PHASE VI - NESSIE AND BIGFOOT',
    'AMAZON BRAZIL', # no match in list, used address
    'APOLLO BLDG (AMAZON)',
    '202 WESTLAKE',
    'SUPPLY LAUNDRY BUILDING', # this is 477 Yale Ave N (Amazon Stackhouse), same building but they changed the street number
    'DAY 1 TOWER AND SPHERES (Amazon)',
    'YALE & THOMAS',
    'RAINIER TOWER',
    'TILT 49 (OFFICE PORTION)',
    '300 PINE STREET',
    'AMAZON-UNICORN',
    '5TH & BELL BUILDING',
    '635 ELLIOTT - SOUTH BUILDING',
    '224 WESTLAKE',
    'SEATTLE CHILDREN\'S HOSPITAL RESEARCH BUILDING #2'
]

amazon_ose_ids += list(all_buildings_dataframe[all_buildings_dataframe['BuildingName'].isin(matched_names)]['OSEBuildingID'])

### Data center at the Westin (used by Amazon)

In [19]:
all_buildings_dataframe[all_buildings_dataframe['BuildingName'].str.contains('WESTIN', case=False)]

OSEBuildingID TaxParcelIdentificationNumber  DataYear    BuildingType  \
56                3                     659000475      2020  NonResidential   
2438            753                     659000950      2020  NonResidential   

                               BuildingName                             Owner  \
56           WESTIN HOTEL (Parent Building)  SEATTLE HOST HOTEL COMPANY L L C   
2438  THE WESTIN BUILDING EXCHANGE (Office)                    2001 SIXTH LLC   

      CouncilDistrictCode Neighborhood      Units  YearBuilt  ...  \
56                      1     DOWNTOWN        891       1969  ...   
2438                    1     DOWNTOWN  NOT_FOUND       1981  ...   

     SecondLargestPropertyUseType SecondLargestPropertyUseTypeGFA  \
56                        Parking                          120545   
2438                       Office                          189244   

     ThirdLargestPropertyUseType  ThirdLargestPropertyUseTypeGFA  Outlier  \
56                 Swimming Pool                               0     None   
2438                     Parking                          169000     None   

      ComplianceIssue ComplianceStatus Comments DefaultData  \
56           No Issue        Compliant     None        None   
2438         No Issue        Compliant     None        None   

      LegislationPropertyType  
56                      Hotel  
2438       Mixed Use Property  

[2 rows x 54 columns]

In [20]:
amazon_ose_ids += [753]


In [21]:
amazon_non_grocery_ose_ids = list(set(amazon_ose_ids))

In [22]:
amazon_non_grocery_buildings = all_buildings_dataframe[all_buildings_dataframe['OSEBuildingID'].isin(amazon_non_grocery_ose_ids)]

amazon_non_grocery_buildings.head()

OSEBuildingID TaxParcelIdentificationNumber  DataYear    BuildingType  \
2            50160                     659000775      2020  NonResidential   
4            50192                     660001605      2020  NonResidential   
91             336                     660000650      2020  NonResidential   
103            388                    7137830040      2020  NonResidential   
125            433                    8634230000      2020  NonResidential   

                BuildingName                          Owner  \
2    AMAZON DOPPLER BUILDING          ACORN DEVELOPMENT LLC   
4        MIDTOWN 21 (AMAZON)                  MIDTOWN21 LLC   
91           1918 EIGHTH AVE  HUDSON 1918 EIGHTH AVENUE LLC   
103            RAINIER TOWER                           None   
125          300 PINE STREET                           None   

     CouncilDistrictCode Neighborhood      Units  YearBuilt  ...  \
2                      7     DOWNTOWN       None       2016  ...   
4                      1     DOWNTOWN       None       2016  ...   
91                     1     DOWNTOWN  NOT_FOUND       2009  ...   
103                    1     DOWNTOWN  NOT_FOUND       1976  ...   
125                    1     DOWNTOWN  NOT_FOUND       1929  ...   

    SecondLargestPropertyUseType SecondLargestPropertyUseTypeGFA  \
2                        Parking                          448625   
4                        Parking                          110813   
91                       Parking                          161119   
103                         None                            <NA>   
125                 Retail Store                          283082   

    ThirdLargestPropertyUseType  ThirdLargestPropertyUseTypeGFA  Outlier  \
2             Convention Center                           59672     None   
4                          None                            <NA>     None   
91                         None                            <NA>     None   
103                        None                            <NA>     None   
125                        None                            <NA>     None   

     ComplianceIssue ComplianceStatus Comments DefaultData  \
2           No Issue        Compliant     None        None   
4           No Issue        Compliant     None        None   
91          No Issue        Compliant     None        None   
103         No Issue        Compliant     None        None   
125         No Issue        Compliant     None        None   

     LegislationPropertyType  
2                     Office  
4                     Office  
91                    Office  
103                   Office  
125                   Office  

[5 rows x 54 columns]

### Whole Foods stores, Amazon Go and Fresh stores

PROBLEM, PROBLEM--none of this adds up because our electricity is carbon neutral. Dead end. See below.

Average polluting of a grocery store: https://www.epa.gov/sites/default/files/documents/gc_averagestoreprofile_final_june_2011_revised_1.pdf

Average EUI for a grocery store: https://portfoliomanager.energystar.gov/pdf/reference/US%20National%20Median%20Table.pdf

Grocery stores are terrible, but Amazon is ranked 2nd best here: https://www.climatefriendlysupermarkets.org/scorecard

Address, name, OSE ID

#### Amazon Go stores

- 2131 7th Ave, Seattle, WA 98121 -- this is in the Spheres building
- 920 5th Ave, Seattle, WA 98104 -- MADISON CENTER (37 floors), 50096
- 1122 Madison St, Seattle, WA 98104 -- THE PERRY- FIRST HILL TOWER (17 floors), 50476
- 300 Boren Ave N, Seattle, WA 98109 -- TROY BLOCK (AMAZON) (so already in list)
- 1906 Terry Ave, Seattle, WA 98101 -- not included in dataset

#### Amazon Fresh stores

- 610 E Pike St, Seattle, WA 98122 -- this is the Ava apartments building in Capitol Hill. Not in dataset, very odd.
- 2301 S Jackson St, Seattle, WA 98144 -- not in dataset
- 76 S Lander St, Seattle, WA 98134 -- STARBUCKS (SODO) CENTER - Campus (maybe 1/7 of the first floor? 9 floors), 640 
- 13201 Aurora Ave N, Seattle, WA 98133 -- AURORA SHOPPING CENTER, 516
- 5100 15th Ave NW, Seattle, WA 98107 -- not in dataset

#### Whole Foods stores

- 2210 Westlake Ave, Seattle, WA 98121 -- not in dataset
- 1001 Broadway, Seattle, WA 98122 -- THE DANFORTH (grocery is 44525 sq ft, total sq ft 592484, ~1/13), 50460
- 2001 15th Ave W, Seattle, WA 98119 -- WHOLE FOODS INTERBAY (whole property is grocery), 700
- 1026 NE 64th St, Seattle, WA 98115 -- ROOSEVELT SQUARE (2 floors, probably about 2/3 of first floor, but 2nd floor only covers 1/3 of first floor, so 1/2), 132
- 4755 Fauntleroy Way SW Ste 190, Seattle, WA 98116 -- THE WHITTAKER, 50259 (7 stories)

In [ ]:
Leaving this aside for the moment, because I'm not convinced it's really reliable to estimate the GHGE from them

In [23]:
# These numbers are deliberately conservative
# ['storeOSEid', 'segments'], floor area calculated as 1/segments

groceries_and_proportions = [
    [50096, 37, 'Grocery - Amazon Go'],
    [50476, 17, 'Grocery - Amazon Go'],
    [640, 63, 'Grocery - Amazon Fresh'],
    [516, 1, 'Grocery - Amazon Fresh'],
    [50460, 13, 'Grocery - Whole Foods'],
    [700, 1, 'Grocery - Whole Foods'],
    [132, 2,'Grocery - Whole Foods'],
    [50259, 7, 'Grocery - Whole Foods']
]

In [24]:
groceries = []

grocery_headers = [
    'OSEBuildingID', 
    'TaxParcelIdentificationNumber',
    'Address',
    'PropertyGFATotal',
    'TotalGHGEmissions__metric_tons_',
    'PortionOfTotalBuilding',
    'StoreGFATotal',
#     'StoreTotalGHGEmissions__metric_tons',
    'SiteEnergyUse_kBtu_',
#     'StoreTotalSiteEnergyUse_kBtu',
    'NaturalGas_kBtu_',
    'Electricity_kBtu_',
    'SteamUse_kBtu_'
]

In [25]:
groceries = []
for store in groceries_and_proportions:
    data = all_buildings_dataframe[all_buildings_dataframe['OSEBuildingID'] == store[0]].iloc[0]
    groceries.append([
        data['OSEBuildingID'], # 'OSEBuildingID', 
        data['TaxParcelIdentificationNumber'], #'TaxParcelIdentificationNumber'
        data['Address'], # 'Address'
        data['PropertyGFATotal'], #'Address'
        data['TotalGHGEmissions__metric_tons_'], # 'TotalGHGEmissions__metric_tons_'
        store[1], # 'PortionOfTotalBuilding'
        data['PropertyGFATotal'] / float(store[1]),
#         data['TotalGHGEmissions__metric_tons_'] / float(store[1]),
        data['SiteEnergyUse_kBtu_'],
#         data['SiteEnergyUse_kBtu_'] / float(store[1]),
        data['NaturalGas_kBtu_'],
        data['Electricity_kBtu_'],
        data['SteamUse_kBtu_']
    ])

In [26]:
grocery_df = pd.DataFrame(data=groceries, columns=grocery_headers)

In [27]:
grocery_df['Owner'] = 'AMAZON'
amazon_grocery_df = grocery_df #because grocery_gf is a terrible name
grocery_df.head()

OSEBuildingID TaxParcelIdentificationNumber             Address  \
0          50096                     942000415         920 5TH AVE   
1          50476                    1978200610      1001 MINOR AVE   
2            640                    7666207220     2401 UTAH AVE S   
3            516                    1926049215  13201 AURORA AVE N   
4          50460                    6390080000      1425 SPRING ST   

   PropertyGFATotal  TotalGHGEmissions__metric_tons_  PortionOfTotalBuilding  \
0            990000                            270.6                      37   
1            235046                            165.7                      17   
2           1750319                            601.3                      63   
3             87665                             24.9                       1   
4            592484                            203.9                      13   

   StoreGFATotal  SiteEnergyUse_kBtu_  NaturalGas_kBtu_  Electricity_kBtu_  \
0   26756.756757             23567748           3084493           20080978   
1   13826.235294              6974812           2708180            4109150   
2   27782.841270             67454992           5098413           62160908   
3   87665.000000              4690846                 0            4690845   
4   45575.692308              8849064           3277350            5620096   

   SteamUse_kBtu_   Owner  
0               0  AMAZON  
1               0  AMAZON  
2               0  AMAZON  
3               0  AMAZON  
4               0  AMAZON

In [28]:
# conservative estimate of GHGE in metric tons
# based on EPA: https://www.epa.gov/sites/default/files/documents/gc_averagestoreprofile_final_june_2011_revised_1.pdf

def estimate_ghge_for_grocery(sq_ft):
    avg_co2_fac = 1383 
    avg_ref_fac = 1556
    return ((sq_ft / 46000.0) * avg_co2_fac) + ((sq_ft / 46000) * avg_ref_fac)

In [137]:
# grocery_df['StoreNaturalGas_kBtu'] = grocery_df['NaturalGas_kBtu_'] / grocery_df['PortionOfTotalBuilding']
# grocery_df['StoreElectricity_kBtu'] = grocery_df['Electricity_kBtu_'] / grocery_df['PortionOfTotalBuilding']
# grocery_df['StoreSteamUse_kBtu'] = grocery_df['SteamUse_kBtu_'] / grocery_df['PortionOfTotalBuilding']

In [29]:
grocery_df['EstStoreGHGE_metric_tons'] = estimate_ghge_for_grocery(grocery_df['StoreGFATotal'])

In [30]:
grocery_df.head()

OSEBuildingID TaxParcelIdentificationNumber             Address  \
0          50096                     942000415         920 5TH AVE   
1          50476                    1978200610      1001 MINOR AVE   
2            640                    7666207220     2401 UTAH AVE S   
3            516                    1926049215  13201 AURORA AVE N   
4          50460                    6390080000      1425 SPRING ST   

   PropertyGFATotal  TotalGHGEmissions__metric_tons_  PortionOfTotalBuilding  \
0            990000                            270.6                      37   
1            235046                            165.7                      17   
2           1750319                            601.3                      63   
3             87665                             24.9                       1   
4            592484                            203.9                      13   

   StoreGFATotal  SiteEnergyUse_kBtu_  NaturalGas_kBtu_  Electricity_kBtu_  \
0   26756.756757             23567748           3084493           20080978   
1   13826.235294              6974812           2708180            4109150   
2   27782.841270             67454992           5098413           62160908   
3   87665.000000              4690846                 0            4690845   
4   45575.692308              8849064           3277350            5620096   

   SteamUse_kBtu_   Owner  EstStoreGHGE_metric_tons  
0               0  AMAZON               1709.524089  
1               0  AMAZON                883.376207  
2               0  AMAZON               1775.081967  
3               0  AMAZON               5601.031196  
4               0  AMAZON               2911.890428

In [31]:
# total emissions for grocery stores
grocery_df['EstStoreGHGE_metric_tons'].sum()

30237.118187121858

In [32]:
aggregate_landlords_df.head()

Unnamed: 0                             Owner  BuildingsOwned  \
0           0                   CITY OF SEATTLE             175   
1           1          UNIVERSITY OF WASHINGTON              43   
2           2               STATE OF WASHINGTON              17   
3           3                       KING COUNTY              16   
4           4  OVERLOOK MAGNOLIA APARTMENTS LLC              11   

   TotalSquareFootage  TotalGHGEmissions__metric_tons_  \
0            17038324                            16357   
1            18445870                            39712   
2             2768746                             3614   
3             4355059                            22535   
4              388815                               52   

   TotalElectricity_kBtu_  TotalSteamUse_kBtu  TotalNaturalGas_kBtu  \
0               501287744             6826192             247129120   
1               679730875           366595771             106707401   
2                88426130                   0              59208493   
3               255701627            16515088             372904431   
4                 9911319                   0                     0   

   TotalOtherFuelUse_kBtu  AverageGHGEmissionsIntensity  \
0                       0                      1.172674   
1                       0                      4.756098   
2                       0                      2.112500   
3                       0                      1.750000   
4                       0                      0.127273   

   AverageSiteEUI_kBtu_sf  AverageSourceEUI_kBtu_sf  AverageENERGYSTARScore  
0               49.970175                108.376608               72.747899  
1              147.265000                321.932500               61.818182  
2               71.162500                134.625000               55.375000  
3               66.943750                137.293750               68.083333  
4               25.418182                 71.145455               74.545455

In [ ]:
Other supermarkets:
QUALITY FOOD CENTERS
SAFEWAY

### Amazon vs. Other Landlords

In [33]:
amazon_v_landlords = aggregate_landlords_df[['Owner', 'BuildingsOwned', 'TotalSquareFootage', 'TotalGHGEmissions__metric_tons_', 'TotalNaturalGas_kBtu', 'AverageGHGEmissionsIntensity']]

In [34]:
amazon_v_landlords.loc[len(amazon_v_landlords)] = [
    'AMAZON OFFICES',
    len(amazon_non_grocery_buildings),
    amazon_non_grocery_buildings['PropertyGFATotal'].sum(),
    amazon_non_grocery_buildings['TotalGHGEmissions__metric_tons_'].sum(),
    amazon_non_grocery_buildings['NaturalGas_kBtu_'].sum(),
    amazon_non_grocery_buildings['GHGEmissionsIntensity'].mean()
]

/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_99492/278652120.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  amazon_v_landlords.loc[len(amazon_v_landlords)] = [


In [35]:
amazon_v_landlords.loc[len(amazon_v_landlords)] = [
    'AMAZON GROCERY',
    len(grocery_df),
    grocery_df['StoreGFATotal'].sum(),
    grocery_df['EstStoreGHGE_metric_tons'].sum(),
    0,
    (grocery_df['EstStoreGHGE_metric_tons'].sum() / grocery_df['StoreGFATotal'].sum()) * 1000
]

/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_99492/393165300.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  amazon_v_landlords.loc[len(amazon_v_landlords)] = [


In [36]:
amazon_v_landlords.loc[len(amazon_v_landlords)] = [
    'ALL AMAZON',
    len(grocery_df) + len(amazon_non_grocery_buildings),
    grocery_df['StoreGFATotal'].sum() + amazon_non_grocery_buildings['PropertyGFATotal'].sum(),
    grocery_df['EstStoreGHGE_metric_tons'].sum() + amazon_non_grocery_buildings['TotalGHGEmissions__metric_tons_'].sum(),
    amazon_non_grocery_buildings['NaturalGas_kBtu_'].sum(),
    (amazon_non_grocery_buildings['GHGEmissionsIntensity'].mean() + ((grocery_df['EstStoreGHGE_metric_tons'].sum() / grocery_df['StoreGFATotal'].sum()) * 1000)) / 2
]

/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_99492/2509588267.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  amazon_v_landlords.loc[len(amazon_v_landlords)] = [


In [37]:
amazon_v_landlords.sort_values('TotalGHGEmissions__metric_tons_', ascending=False)[:20]

Owner  BuildingsOwned  \
1                          UNIVERSITY OF WASHINGTON              43   
2080                                     ALL AMAZON              40   
2079                                 AMAZON GROCERY               8   
3                                       KING COUNTY              16   
0                                   CITY OF SEATTLE             175   
152                  SWEDISH HOSPITAL MEDICAL CENTE               2   
2078                                 AMAZON OFFICES              32   
896                                 JAMES TOWER LLC               1   
1706                      KOLL LAKE UNION ASSOC L P               1   
2                               STATE OF WASHINGTON              17   
14                               SEATTLE UNIVERSITY               6   
17                       SEATTLE PACIFIC UNIVERSITY               5   
1445  CHILDREN'S HOSPITAL & REGIONAL MEDICAL CENTER               1   
1602                       CITY INVESTORS XII L L C               1   
1446                           HT-SEATTLE OWNER LLC               1   
51                  UNIVERSITY OF HAWAII FOUNDATION               2   
10                            ACORN DEVELOPMENT LLC               9   
1376             WASHINGTON STATE CONVENTION CENTER               1   
310                                  2001 SIXTH LLC               1   
1237                            HART FIRST HILL LLC               1   

      TotalSquareFootage  TotalGHGEmissions__metric_tons_  \
1           1.844587e+07                     39712.000000   
2080        1.657504e+07                     36622.218187   
2079        4.732587e+05                     30237.118187   
3           4.355059e+06                     22535.000000   
0           1.703832e+07                     16357.000000   
152         1.659904e+06                     11941.000000   
2078        1.610178e+07                      6385.100000   
896         5.975190e+05                      4805.000000   
1706        5.272890e+05                      4524.000000   
2           2.768746e+06                      3614.000000   
14          2.624288e+06                      3019.000000   
17          7.538010e+05                      2150.000000   
1445        2.306380e+05                      1927.000000   
1602        2.255810e+05                      1707.000000   
1446        1.062251e+06                      1620.000000   
51          3.850480e+05                      1619.000000   
10          5.109919e+06                      1595.000000   
1376        9.462120e+05                      1471.000000   
310         4.294050e+05                      1455.000000   
1237        2.275690e+05                      1439.000000   

      TotalNaturalGas_kBtu  AverageGHGEmissionsIntensity  
1                106707401                      4.756098  
2080              40783705                     32.194090  
2079                     0                     63.891304  
3                372904431                      1.750000  
0                247129120                      1.172674  
152               16108152                      5.600000  
2078              40783705                      0.496875  
896               84872338                      8.000000  
1706              79160244                      8.600000  
2                 59208493                      2.112500  
14                51904014                      0.533333  
17                38279666                      1.500000  
1445              33574922                      8.400000  
1602              28404272                      7.600000  
1446              27548291                      1.500000  
51                26048330                      4.900000  
10                15879116                      0.177778  
1376               1251147                      1.600000  
310                 315739                      3.400000  
1237                     0                      6.300000

Amazon as a whole is the 2nd largest emitter, ~36k metric tons of CO2/yr. UW not far ahead at 39k.

Amazon grocery is 2nd if we separate offices and grocery, 30k. And grocery at #6, 6k tons.

If we exclude grocery, then we could say Amazon's the 5th worse emitter. But the problem is that we're looking at a drop from 39k with UW to 6k with Amazon offices. That's pretty pathetic.
That's still the worst office buildings, minus the city and state. So heaviest polluting corporate offices. That's something we could work with.

In [178]:
amazon_v_landlords.sort_values('TotalSquareFootage', ascending=False)[:10]

Owner  BuildingsOwned  \
1                   UNIVERSITY OF WASHINGTON              43   
0                            CITY OF SEATTLE             175   
2080                              ALL AMAZON              40   
2078                          AMAZON OFFICES              32   
10                     ACORN DEVELOPMENT LLC               9   
3                                KING COUNTY              16   
2                        STATE OF WASHINGTON              17   
14                        SEATTLE UNIVERSITY               6   
136   UNION SQUARE LIMITED LIABILITY COMPANY               2   
50       WASHINGTON ST PUBLIC STADIUM ATHRTY               2   

      TotalSquareFootage  TotalGHGEmissions__metric_tons_  \
1           1.844587e+07                     39712.000000   
0           1.703832e+07                     16357.000000   
2080        1.657504e+07                     36622.218187   
2078        1.610178e+07                      6385.100000   
10          5.109919e+06                      1595.000000   
3           4.355059e+06                     22535.000000   
2           2.768746e+06                      3614.000000   
14          2.624288e+06                      3019.000000   
136         1.973704e+06                       561.000000   
50          1.919658e+06                       824.000000   

      TotalNaturalGas_kBtu  AverageGHGEmissionsIntensity  
1                106707401                      4.756098  
0                247129120                      1.172674  
2080              40783705                     32.194090  
2078              40783705                      0.496875  
10                15879116                      0.177778  
3                372904431                      1.750000  
2                 59208493                      2.112500  
14                51904014                      0.533333  
136                1899978                      0.250000  
50                10829543                      0.500000

Amazon also the largest by sq ft. That's pretty good. If we estimate more numbers based on othese other buildings not in the data set, we'll get even higher numbers.

Can't include the stadium, people like sports. Boo.

Side note: look at Acorn, what's their deal?

In [302]:
non_amazon_groceries = all_buildings_dataframe[all_buildings_dataframe['EPAPropertyType'] == 'Supermarket/Grocery Store']

# just in case any of them aren't marked as grocery stores
all_buildings_dataframe[
    (all_buildings_dataframe['BuildingName'].str.contains('QFC')) |
    (all_buildings_dataframe['BuildingName'].str.contains('QUALITY FOOD CENTER'))
]['Owner'] = 'QFC'

/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_80789/3107445035.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  all_buildings_dataframe[


OSEBuildingID TaxParcelIdentificationNumber  DataYear  \
1112          20045                     925049330      2020   
1281          49946                    2767604965      2020   
1740          35392                    4458720010      2020   
1971          20432                    1333100000      2020   
2358          22479                    2770602660      2020   
2815            867                    3148650000      2020   
2976          27821                    5724500615      2020   
3308          20613                    1824049069      2020   
3488            802                    5260300005      2020   
3501            558                    6632300290      2020   
3504            381                     767000060      2020   

              BuildingType                           BuildingName Owner  \
1112        NonResidential                  2746 NE 45TH ST - QFC   QFC   
1281        NonResidential                          QFC - Ballard   QFC   
1740        NonResidential                      QFC at 500 MERCER   QFC   
1971  Multifamily MR (5-9)      CAPCO PLAZA (ALTAMIRA + QFC ETC.)   QFC   
2358        NonResidential                             QFC MARKET   QFC   
2815        NonResidential           HARVARD MARKET (QFC Portion)   QFC   
2976        NonResidential  QUALITY FOOD CENTER (QFC) SUPERMARKET   QFC   
3308        NonResidential                          QFC WAREHOUSE   QFC   
3488        NonResidential            QFC AND RITE AID DRUG STORE   QFC   
3501        NonResidential                     QFC & DOLLAR STORE   QFC   
3504        NonResidential                        QFC HOLMAN ROAD   QFC   

      CouncilDistrictCode                   Neighborhood      Units  \
1112                    1                      NORTHEAST  NOT_FOUND   
1281                    1                        BALLARD       None   
1740                    7          MAGNOLIA / QUEEN ANNE       None   
1971                    1                      SOUTHWEST        157   
2358                    7          MAGNOLIA / QUEEN ANNE  NOT_FOUND   
2815                    1                           EAST  NOT_FOUND   
2976                    1  SHARED: NORTH &amp; NORTHWEST  NOT_FOUND   
3308                    2               GREATER DUWAMISH  NOT_FOUND   
3488                    1               GREATER DUWAMISH  NOT_FOUND   
3501                    5                          NORTH  NOT_FOUND   
3504                    1                      NORTHWEST  NOT_FOUND   

      YearBuilt  ... SecondLargestPropertyUseType  \
1112       1955  ...                         None   
1281       2010  ...                         None   
1740       2006  ...                         None   
1971       2008  ...                         None   
2358       1961  ...                         None   
2815       1996  ...                         None   
2976       1996  ...                         None   
3308       1951  ...                         None   
3488       1963  ...                 Retail Store   
3501       1980  ...                         None   
3504       1956  ...                         None   

     SecondLargestPropertyUseTypeGFA ThirdLargestPropertyUseType  \
1112                            <NA>                        None   
1281                            <NA>                        None   
1740                            <NA>                        None   
1971                            <NA>                        None   
2358                            <NA>                        None   
2815                            <NA>                        None   
2976                            <NA>                        None   
3308                            <NA>                        None   
3488                           22081                        None   
3501                            <NA>                        None   
3504                            <NA>                        None   

      ThirdLargestPropertyUseTypeGFA  Out

In [304]:
all_buildings_dataframe[all_buildings_dataframe['BuildingName'].str.contains('SAFEWAY')]['Owner'] = 'SAFEWAY'


/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_80789/2945430338.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  all_buildings_dataframe[all_buildings_dataframe['BuildingName'].str.contains('SAFEWAY')]['Owner'] = 'SAFEWAY'


In [198]:
qfc['EPAPropertyType'].unique()

array(['Supermarket/Grocery Store', 'Unclassified',
       'Non-Refrigerated Warehouse'], dtype=object)

In [199]:
safeway['EPAPropertyType'].unique()

array(['Supermarket/Grocery Store'], dtype=object)

In [207]:
qfc.reset_index(drop=True, inplace=True)
safeway.reset_index(drop=True, inplace=True)
non_amazon_groceries.reset_index(drop=True, inplace=True)

In [230]:
all_non_amazon_grocery = pd.concat([qfc, safeway, non_amazon_groceries])

In [231]:
all_non_amazon_grocery

OSEBuildingID TaxParcelIdentificationNumber  DataYear  \
1112          20045                     925049330      2020   
1281          49946                    2767604965      2020   
1740          35392                    4458720010      2020   
1971          20432                    1333100000      2020   
2358          22479                    2770602660      2020   
...             ...                           ...       ...   
3389          27162                    5637500005      2020   
3488            802                    5260300005      2020   
3489          23098                    3086003145      2020   
3501            558                    6632300290      2020   
3504            381                     767000060      2020   

              BuildingType                       BuildingName  \
1112        NonResidential              2746 NE 45TH ST - QFC   
1281        NonResidential                      QFC - Ballard   
1740        NonResidential                  QFC at 500 MERCER   
1971  Multifamily MR (5-9)  CAPCO PLAZA (ALTAMIRA + QFC ETC.)   
2358        NonResidential                         QFC MARKET   
...                    ...                                ...   
3389        NonResidential             WEST SEATTLE THRIFTWAY   
3488        NonResidential        QFC AND RITE AID DRUG STORE   
3489        NonResidential           HILLTOP RED APPLE MARKET   
3501        NonResidential                 QFC & DOLLAR STORE   
3504        NonResidential                    QFC HOLMAN ROAD   

                           Owner  CouncilDistrictCode           Neighborhood  \
1112                         QFC                    1              NORTHEAST   
1281                         QFC                    1                BALLARD   
1740                         QFC                    7  MAGNOLIA / QUEEN ANNE   
1971                         QFC                    1              SOUTHWEST   
2358                         QFC                    7  MAGNOLIA / QUEEN ANNE   
...                          ...                  ...                    ...   
3389       JORVE INVESTMENTS LLC                    1              SOUTHWEST   
3488        RAINIER PLAZA #2 LLC                    1       GREATER DUWAMISH   
3489       METROPOLE HILLTOP LLC                    2       GREATER DUWAMISH   
3501  ABC PACIFIC CORPORATION $T                    5                  NORTH   
3504                    FRMC LLC                    1              NORTHWEST   

          Units  YearBuilt  ... SecondLargestPropertyUseType  \
1112  NOT_FOUND       1955  ...                         None   
1281       None       2010  ...                         None   
1740       None       2006  ...                         None   
1971        157       2008  ...                         None   
2358  NOT_FOUND       1961  ...                         None   
...         ...        ...  ...                          ...   
3389  NOT_FOUND       1998  ...                         None   
3488  NOT_FOUND       1963  ...                 Retail Store   
3489  NOT_FOUND       1965  ...                         None   
3501  NOT_FOUND       1980  ...                         None   
3504  NOT_FOUND       1956  ...                         None   

     SecondLargestPropertyUseTypeGFA ThirdLargestPropertyUseType  \
1112                            <NA>                        None   
1281                            <NA>                        None   
1740                            <NA>                        None   
1971                            <NA>                        None   
2358                            <NA>                        None   
...                              ...                         ...   
3389                            <NA>                        None   
3488                           22081                        None   
3489                            <NA>                        None   
3501                            <NA>                      

In [232]:
all_non_amazon_grocery.drop_duplicates(subset='OSEBuildingID', inplace=True)

In [269]:
grocery_by_ghge = all_non_amazon_grocery[['Owner', 'Address', 'TotalGHGEmissions__metric_tons_']]
amazon_grocery_df['TotalGHGEmissions__metric_tons_'] = amazon_grocery_df['EstStoreGHGE_metric_tons'] # normalize column names
amazon_grocery_df.head()

OSEBuildingID TaxParcelIdentificationNumber             Address  \
0          50096                     942000415         920 5TH AVE   
1          50476                    1978200610      1001 MINOR AVE   
2            640                    7666207220     2401 UTAH AVE S   
3            516                    1926049215  13201 AURORA AVE N   
4          50460                    6390080000      1425 SPRING ST   

   PropertyGFATotal  TotalGHGEmissions__metric_tons_  PortionOfTotalBuilding  \
0            990000                      1709.524089                      37   
1            235046                       883.376207                      17   
2           1750319                      1775.081967                      63   
3             87665                      5601.031196                       1   
4            592484                      2911.890428                      13   

   StoreGFATotal  SiteEnergyUse_kBtu_  NaturalGas_kBtu_  Electricity_kBtu_  \
0   26756.756757             23567748           3084493           20080978   
1   13826.235294              6974812           2708180            4109150   
2   27782.841270             67454992           5098413           62160908   
3   87665.000000              4690846                 0            4690845   
4   45575.692308              8849064           3277350            5620096   

   SteamUse_kBtu_  EstStoreGHGE_metric_tons   Owner  
0               0               1709.524089  AMAZON  
1               0                883.376207  AMAZON  
2               0               1775.081967  AMAZON  
3               0               5601.031196  AMAZON  
4               0               2911.890428  AMAZON

In [273]:
# grocery_by_ghge[grocery_by_ghge['Owner'] == 'AMAZON']['TotalGHGEmissions__metric_tons_'] = grocery_by_ghge['EstStoreGHGE_metric_tons']
all_grocery_ghge = pd.concat([amazon_grocery_df[['Owner', 'Address', 'TotalGHGEmissions__metric_tons_']], grocery_by_ghge])


In [274]:
grocery_by_ghge_ranking = all_grocery_ghge.groupby('Owner')['TotalGHGEmissions__metric_tons_'].sum().sort_values(ascending=False)
grocery_by_ghge_ranking

Owner
AMAZON                                                    30237.118187
SAFEWAY                                                    3056.700000
QFC                                                        1773.900000
COSTCO WHOLESALE CORPORATION                                529.000000
FRED MEYER STORES INC                                       423.000000
GREENWOOD FM PROPERTIES LLC                                 365.200000
ADMIRAL FARM TO MARKET L L C                                356.800000
PAJIS WA L L C                                              321.100000
DS INTERBAY URBAN CENTER LP                                 291.200000
RRDS LLC                                                    284.000000
DAMBROSIO ENTERPRISES LLC                                   244.700000
ROOSEVELT SQUARE REGENCY LLC                                145.200000
OAKTREE PLAZA ASSOCIATES                                    134.400000
METROPOLE HILLTOP LLC                                        96.500000


In [291]:
all_non_amazon_grocery['StoreGFATotal'] = all_non_amazon_grocery['PropertyGFABuilding_s_'] #normalize columns
grocery_size_ranking = pd.concat([
    amazon_grocery_df[['Owner', 'Address', 'TotalGHGEmissions__metric_tons_', 'StoreGFATotal']], 
    all_non_amazon_grocery[['Owner', 'Address', 'TotalGHGEmissions__metric_tons_', 'StoreGFATotal']]
])

In [280]:
grocery_size_ranking.head()

Owner             Address  TotalGHGEmissions__metric_tons_  StoreGFATotal
0  AMAZON         920 5TH AVE                      1709.524089   26756.756757
1  AMAZON      1001 MINOR AVE                       883.376207   13826.235294
2  AMAZON     2401 UTAH AVE S                      1775.081967    27782.84127
3  AMAZON  13201 AURORA AVE N                      5601.031196        87665.0
4  AMAZON      1425 SPRING ST                      2911.890428   45575.692308

In [292]:
# df.groupby(['col1', 'col2'])["col3", "col4"].apply(lambda x : x.astype(int).sum())

grocery_size_ranking = grocery_size_ranking.groupby(['Owner'])['StoreGFATotal', 'TotalGHGEmissions__metric_tons_'].apply(lambda x: x.sum())
#.sort_values('StoreGFATotal', ascending=False, inplace=True)


/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_80789/2654747329.py:3: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  grocery_size_ranking = grocery_size_ranking.groupby(['Owner'])['StoreGFATotal', 'TotalGHGEmissions__metric_tons_'].apply(lambda x: x.sum())


StoreGFATotal  \
Owner                                                               
1550 ARMORY LLC                                      25065.000000   
ADMIRAL FARM TO MARKET L L C                         31841.000000   
AMAZON                                              473258.739914   
ANYU LLC (89.4%)+KREMER FAMILY REVOCABLE TRUST ...   50083.000000   
COSTCO WHOLESALE CORPORATION                        158415.000000   
DAMBROSIO ENTERPRISES LLC                            28808.000000   
DS INTERBAY URBAN CENTER LP                          64659.000000   
FRED MEYER STORES INC                               121087.000000   
GREENWOOD FM PROPERTIES LLC                         118133.000000   
HULING BROS PROP LLC                                 25762.000000   
JORVE INVESTMENTS LLC                                33706.000000   
METROPOLE HILLTOP LLC                                21603.000000   
OAKTREE PLAZA ASSOCIATES                             45155.000000   
OB SODO PROPERTIES V LLC                             23792.000000   
PAJIS WA L L C                                       24377.000000   
QFC                                                 653745.000000   
ROOSEVELT SQUARE REGENCY LLC                        128315.000000   
RRDS LLC                                             29214.000000   
SAFEWAY                                             497657.000000   
SAINT GERVAIS LLC % STARRETT CTY ASSO                67450.000000   

                                                    TotalGHGEmissions__metric_tons_  
Owner                                                                                
1550 ARMORY LLC                                                            6.900000  
ADMIRAL FARM TO MARKET L L C                                             356.800000  
AMAZON                                                                 30237.118187  
ANYU LLC (89.4%)+KREMER FAMILY REVOCABLE TRUST ...                        15.200000  
COSTCO WHOLESALE CORPORATION                                             529.000000  
DAMBROSIO ENTERPRISES LLC                                                244.700000  
DS INTERBAY URBAN CENTER LP                                              291.200000  
FRED MEYER STORES INC                                                    423.000000  
GREENWOOD FM PROPERTIES LLC                                              365.200000  
HULING BROS PROP LLC                                                      10.000000  
JORVE INVESTMENTS LLC                                                      0.000000  
METROPOLE HILLTOP LLC                                                     96.500000  
OAKTREE PLAZA ASSOCIATES                                                 134.400000  
OB SODO PROPERTIES V LLC                                                  13.400000  
PAJIS WA L L C                                                           321.100000  
QFC                                                                     1773.900000  
ROOSEVELT SQUARE REGENCY LLC                                             145.200000  
RRDS LLC                                                                 284.000000  
SAFEWAY                                                                 3056.700000  
SAINT GERVAIS LLC % STARRETT CTY ASSO                                     42.600000

In [293]:
grocery_size_ranking.sort_values('StoreGFATotal', ascending=False, inplace=True)

In [294]:
grocery_size_ranking

StoreGFATotal  \
Owner                                                               
QFC                                                 653745.000000   
SAFEWAY                                             497657.000000   
AMAZON                                              473258.739914   
COSTCO WHOLESALE CORPORATION                        158415.000000   
ROOSEVELT SQUARE REGENCY LLC                        128315.000000   
FRED MEYER STORES INC                               121087.000000   
GREENWOOD FM PROPERTIES LLC                         118133.000000   
SAINT GERVAIS LLC % STARRETT CTY ASSO                67450.000000   
DS INTERBAY URBAN CENTER LP                          64659.000000   
ANYU LLC (89.4%)+KREMER FAMILY REVOCABLE TRUST ...   50083.000000   
OAKTREE PLAZA ASSOCIATES                             45155.000000   
JORVE INVESTMENTS LLC                                33706.000000   
ADMIRAL FARM TO MARKET L L C                         31841.000000   
RRDS LLC                                             29214.000000   
DAMBROSIO ENTERPRISES LLC                            28808.000000   
HULING BROS PROP LLC                                 25762.000000   
1550 ARMORY LLC                                      25065.000000   
PAJIS WA L L C                                       24377.000000   
OB SODO PROPERTIES V LLC                             23792.000000   
METROPOLE HILLTOP LLC                                21603.000000   

                                                    TotalGHGEmissions__metric_tons_  
Owner                                                                                
QFC                                                                     1773.900000  
SAFEWAY                                                                 3056.700000  
AMAZON                                                                 30237.118187  
COSTCO WHOLESALE CORPORATION                                             529.000000  
ROOSEVELT SQUARE REGENCY LLC                                             145.200000  
FRED MEYER STORES INC                                                    423.000000  
GREENWOOD FM PROPERTIES LLC                                              365.200000  
SAINT GERVAIS LLC % STARRETT CTY ASSO                                     42.600000  
DS INTERBAY URBAN CENTER LP                                              291.200000  
ANYU LLC (89.4%)+KREMER FAMILY REVOCABLE TRUST ...                        15.200000  
OAKTREE PLAZA ASSOCIATES                                                 134.400000  
JORVE INVESTMENTS LLC                                                      0.000000  
ADMIRAL FARM TO MARKET L L C                                             356.800000  
RRDS LLC                                                                 284.000000  
DAMBROSIO ENTERPRISES LLC                                                244.700000  
HULING BROS PROP LLC                                                      10.000000  
1550 ARMORY LLC                                                            6.900000  
PAJIS WA L L C                                                           321.100000  
OB SODO PROPERTIES V LLC                                                  13.400000  
METROPOLE HILLTOP LLC                                                     96.500000

##### Whoa, are the other supermarkets under-reporting??

In [296]:
# back of the envelope for Amazon
# our numbers for amazon were calculated with electric and r-404a emissions: https://www.epa.gov/sites/default/files/documents/gc_averagestoreprofile_final_june_2011_revised_1.pdf
# maybe the other grocers didn't include that?

avg_supermarket = 46000
metric_tons_co2_for_avg_supermarket = 1383
num_avg_amazon_markets = 473258 / avg_supermarket
amazon_markets_ghge = num_avg_amazon_markets * metric_tons_co2_for_avg_supermarket
print('Estimated electric-only GHGE impact of all Amazon supermarkets')
print(amazon_markets_ghge)

Estimated electric-only GHGE impact of all Amazon supermarkets
14228.604652173914


Amazon has nearly five times the GHGE for its grocery stores compared to other grocery stores (counting electricity only)

Let's look at QFC, the largest grocery by total area:

In [297]:
num_avg_qfc_markets = 653745 / avg_supermarket
qfc_markets_ghge = num_avg_qfc_markets * metric_tons_co2_for_avg_supermarket
print('Estimated electric-only GHGE impact of all QFC supermarkets')
print(qfc_markets_ghge)

Estimated electric-only GHGE impact of all QFC supermarkets
19654.98554347826


So QFC would be expected to produce at least 20k metric tons of GHGE from electricity alone. And yet they produce less than 2k, allegedly!

In [308]:
all_non_amazon_grocery[all_non_amazon_grocery['Owner'] == 'QFC']['TotalGHGEmissions__metric_tons_'].sum()

1773.9

Am I imagining things? This is too low. Let's look at the exact numbers:

In [319]:
print('Metric tons GHGE from QFC stores')
all_non_amazon_grocery[all_non_amazon_grocery['Owner'] == 'QFC']['TotalGHGEmissions__metric_tons_']

Metric tons GHGE from QFC stores


1112    306.7
1281     50.3
1740    132.0
1971      0.0
2358    223.5
2815    189.1
2976    219.2
3308     38.5
3488    175.2
3501    158.8
3504    280.6
Name: TotalGHGEmissions__metric_tons_, dtype: float64

Seattle City Light is clean electricity, so in theory there should be no emissions from electricity. Given that, let's look at gas and steam use.

In [322]:
print('Gas and steam use for QFC stores')
all_non_amazon_grocery[all_non_amazon_grocery['Owner'] == 'QFC'][['NaturalGas_kBtu_', 'SteamUse_kBtu_']]

Gas and steam use for QFC stores


NaturalGas_kBtu_  SteamUse_kBtu_
1112           4904319               0
1281            489591               0
1740           2007857               0
1971                 0               0
2358           3767896               0
2815           3064567               0
2976           3617811               0
3308            689301               0
3488           2789475               0
3501           2467165               0
3504           4623164               0

So does gas explain it? Let's see how much CO2 would be produced by this amount of gas.

In [323]:
total_gas_therms = all_non_amazon_grocery[all_non_amazon_grocery['Owner'] == 'QFC']['NaturalGas_therms_'].sum()
total_gas_therms

284213

Using this [EPA calculator](https://www.epa.gov/energy/greenhouse-gas-equivalencies-calculator#results), 284,213 therms of natural gas produces about 1,504 metric tons of CO2. QFC's reported CO2 is 1700 metric tons, so roughly equivalent (where's the extra 200 from? Unclear). 

(BTW, the calculator is super cool, and can get us fun facts like "this building emits as much co2 as 20 jet planes).

So natural gas accounts for most of the reported CO2 production at QFC. This suggests that the Amazon statistics aren't really going to be that much worse, given that we're estimating them using nationwide averages. They may even be better, since QFC is supposed to be very bad being good to the environment (see link above where it says that Kroger kills the planet). And also Amazon is known for trying to do things that look good in their public-facing spaces (heat transfer from the Westin, etc.) so it wouldn't be surprising if they're trying to be more energy efficient in their stores, though we have no data to back up that conjecture.

Conclusion: supermarkets are a dead end, alas.

## Amazon vs. Other Types of Buildings

### Amazon vs. Other Categories

In [43]:
amazon_non_grocery_buildings['Owner'] = 'AMAZON'

/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_99492/215153507.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  amazon_non_grocery_buildings['Owner'] = 'AMAZON'


In [45]:
# amazon_non_grocery_buildings
# amazon_non_grocery_ose_ids

# grocery_size_ranking = grocery_size_ranking.groupby(['Owner'])['StoreGFATotal', 'TotalGHGEmissions__metric_tons_'].apply(lambda x: x.sum())

non_amazon_buildings = all_buildings_dataframe[~all_buildings_dataframe['OSEBuildingID'].isin(amazon_non_grocery_ose_ids)]

amazon_v_categories = pd.concat([
    non_amazon_buildings.groupby(['EPAPropertyType'])['PropertyGFABuilding_s_', 'TotalGHGEmissions__metric_tons_', 'Electricity_kBtu_', 'NaturalGas_kBtu_', 'SteamUse_kBtu_'].apply(lambda x: x.sum()),
    amazon_non_grocery_buildings.groupby(['Owner'])[['PropertyGFABuilding_s_', 'TotalGHGEmissions__metric_tons_', 'Electricity_kBtu_', 'NaturalGas_kBtu_', 'SteamUse_kBtu_']].apply(lambda x: x.sum())
])

/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_99492/2775666461.py:9: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  non_amazon_buildings.groupby(['EPAPropertyType'])['PropertyGFABuilding_s_', 'TotalGHGEmissions__metric_tons_', 'Electricity_kBtu_', 'NaturalGas_kBtu_', 'SteamUse_kBtu_'].apply(lambda x: x.sum()),


In [46]:
amazon_v_categories.sort_values('TotalGHGEmissions__metric_tons_', ascending=False)[:20]

PropertyGFABuilding_s_  \
Hospital (General Medical & Surgical)               8333776.0   
Multifamily MR (5-9)                               58994840.0   
Mixed Use Property                                 14001717.0   
Office                                             61645488.0   
Multifamily HR (10+)                               26991285.0   
College/University                                 16745460.0   
Multifamily LR (1-4)                               42402787.0   
Hotel                                              11825876.0   
Medical Office                                      5611194.0   
K-12 School                                        11478922.0   
Unclassified                                        9073104.0   
Other                                               3757549.0   
Laboratory                                          1315179.0   
Supermarket/Grocery Store                           1990332.0   
Non-Refrigerated Warehouse                          8663931.0   
AMAZON                                             11912586.0   
Retail Store                                        4348122.0   
Parking                                             2517267.0   
Other - Recreation                                  1264151.0   
Other - Entertainment/Public Assembly               3425419.0   

                                       TotalGHGEmissions__metric_tons_  \
Hospital (General Medical & Surgical)                          58799.7   
Multifamily MR (5-9)                                           44650.4   
Mixed Use Property                                             40280.8   
Office                                                         36449.8   
Multifamily HR (10+)                                           29035.9   
College/University                                             26708.2   
Multifamily LR (1-4)                                           25236.7   
Hotel                                                          18808.1   
Medical Office                                                 14362.5   
K-12 School                                                    12683.8   
Unclassified                                                   11725.2   
Other                                                          11423.7   
Laboratory                                                      9300.7   
Supermarket/Grocery Store                                       8112.5   
Non-Refrigerated Warehouse                                      6598.3   
AMAZON                                                          6385.1   
Retail Store                                                    4428.8   
Parking                                                         3487.4   
Other - Recreation                                              3143.7   
Other - Entertainment/Public Assembly                           2729.3   

                                       Electricity_kBtu_  NaturalGas_kBtu_  \
Hospital (General Medical & Surgical)       7.213609e+08       345057738.0   
Multifamily MR (5-9)                        1.374625e+09       685291586.0   
Mixed Use Property                          1.077977e+09       631426511.0   
Office                                      2.548258e+09       271898626.0   
Multifamily HR (10+)                        7.162368e+08       417235726.0   
College/University                          3.374291e+08       131408620.0   
Multifamily LR (1-4)                        1.018129e+09       372479306.0   
Hotel                                       3.686840e+08       210887900.0   
Medical Office                              3.963446e+08       130663729.0   
K-12 School                                 1.720931e+08       221026130.0   
Unclassified                                1.759921e+08       180457943.0   
Other                                       2.779534e+08       169421151.0   
Laboratory                                  1.505434e+08       117231433.0   
Supermarket/Grocery Store 

In [47]:
amazon_v_categories.to_csv('amazon_v_building_category.csv')

So Amazon ranks 14th for emissions if it's its own category (removing "unclassified" and "other"). Not all that bad, which in turn is bad for us.

In [48]:
amazon_v_categories.sort_values('NaturalGas_kBtu_', ascending=False)[:20]

PropertyGFABuilding_s_  \
Multifamily MR (5-9)                               58994840.0   
Mixed Use Property                                 14001717.0   
Multifamily HR (10+)                               26991285.0   
Multifamily LR (1-4)                               42402787.0   
Hospital (General Medical & Surgical)               8333776.0   
Office                                             61645488.0   
K-12 School                                        11478922.0   
Hotel                                              11825876.0   
Unclassified                                        9073104.0   
Other                                               3757549.0   
College/University                                 16745460.0   
Medical Office                                      5611194.0   
Supermarket/Grocery Store                           1990332.0   
Laboratory                                          1315179.0   
Non-Refrigerated Warehouse                          8663931.0   
Retail Store                                        4348122.0   
Other - Recreation                                  1264151.0   
Parking                                             2517267.0   
Worship Facility                                    2091103.0   
AMAZON                                             11912586.0   

                                       TotalGHGEmissions__metric_tons_  \
Multifamily MR (5-9)                                           44650.4   
Mixed Use Property                                             40280.8   
Multifamily HR (10+)                                           29035.9   
Multifamily LR (1-4)                                           25236.7   
Hospital (General Medical & Surgical)                          58799.7   
Office                                                         36449.8   
K-12 School                                                    12683.8   
Hotel                                                          18808.1   
Unclassified                                                   11725.2   
Other                                                          11423.7   
College/University                                             26708.2   
Medical Office                                                 14362.5   
Supermarket/Grocery Store                                       8112.5   
Laboratory                                                      9300.7   
Non-Refrigerated Warehouse                                      6598.3   
Retail Store                                                    4428.8   
Other - Recreation                                              3143.7   
Parking                                                         3487.4   
Worship Facility                                                2690.3   
AMAZON                                                          6385.1   

                                       Electricity_kBtu_  NaturalGas_kBtu_  \
Multifamily MR (5-9)                        1.374625e+09       685291586.0   
Mixed Use Property                          1.077977e+09       631426511.0   
Multifamily HR (10+)                        7.162368e+08       417235726.0   
Multifamily LR (1-4)                        1.018129e+09       372479306.0   
Hospital (General Medical & Surgical)       7.213609e+08       345057738.0   
Office                                      2.548258e+09       271898626.0   
K-12 School                                 1.720931e+08       221026130.0   
Hotel                                       3.686840e+08       210887900.0   
Unclassified                                1.759921e+08       180457943.0   
Other                                       2.779534e+08       169421151.0   
College/University                          3.374291e+08       131408620.0   
Medical Office                              3.963446e+08       130663729.0   
Supermarket/Grocery Store                   2.245628e+08       130263652.0   
Laboratory                

In [346]:
amazon_v_categories['GHGEIntensityAcrossBuildings'] = amazon_v_categories['TotalGHGEmissions__metric_tons_'] / amazon_v_categories['PropertyGFABuilding_s_']

In [349]:
amazon_v_categories.sort_values('GHGEIntensityAcrossBuildings', ascending=False)[:20]

PropertyGFABuilding_s_  \
Data Center                                                     83029.0   
Laboratory                                                    1315179.0   
Hospital (General Medical & Surgical)                         8333776.0   
Restaurant                                                     214717.0   
Prison/Incarceration                                           385274.0   
Other/Specialty Hospital                                       128399.0   
Supermarket/Grocery Store                                     1990332.0   
Other - Public Services                                         77039.0   
Lifestyle Center                                                81336.0   
Urgent Care/Clinic/Other Outpatient                             29796.0   
Other                                                         3757549.0   
Fitness Center/Health Club/Gym                                 298932.0   
Mixed Use Property                                           14001717.0   
Police Station                                                 123201.0   
Repair Services (Vehicle, Shoe, Locksmith, etc)                328638.0   
Residential Care Facility                                      120852.0   
Museum                                                         942156.0   
Medical Office                                                5611194.0   
Social/Meeting Hall                                            361255.0   
Other - Recreation                                            1264151.0   

                                                 TotalGHGEmissions__metric_tons_  \
Data Center                                                                628.1   
Laboratory                                                                9300.7   
Hospital (General Medical & Surgical)                                    58799.7   
Restaurant                                                                1145.1   
Prison/Incarceration                                                      1671.1   
Other/Specialty Hospital                                                   524.7   
Supermarket/Grocery Store                                                 8112.5   
Other - Public Services                                                    303.2   
Lifestyle Center                                                           288.7   
Urgent Care/Clinic/Other Outpatient                                         97.5   
Other                                                                    11423.7   
Fitness Center/Health Club/Gym                                             862.9   
Mixed Use Property                                                       40280.8   
Police Station                                                             342.5   
Repair Services (Vehicle, Shoe, Locksmith, etc)                            892.2   
Residential Care Facility                                                  324.8   
Museum                                                                    2510.1   
Medical Office                                                           14362.5   
Social/Meeting Hall                                                        914.4   
Other - Recreation                                                        3143.7   

                                                 Electricity_kBtu_  \
Data Center                                           1.181053e+08   
Laboratory                                            1.505434e+08   
Hospital (General Medical & Surgical)                 7.213609e+08   
Restaurant                                            1.638668e+07   
Prison/Incarceration                                  1.838236e+07   
Other/Specialty Hospital                              1.286134e+07   
Supermarket/Grocery Store                             2.245628e+08   
Other - Public Services                               3.751935e+06   
Lifestyle Center                                      8.954589e+06   
Urgent C

Not much luck here, either.

Need to compare vs. legislative categories as well.

In [49]:


amazon_v_landlords = pd.concat([
    non_amazon_buildings.groupby(['Owner'])['PropertyGFABuilding_s_', 'TotalGHGEmissions__metric_tons_', 'Electricity_kBtu_', 'NaturalGas_kBtu_', 'SteamUse_kBtu_'].apply(lambda x: x.sum()),
    amazon_non_grocery_buildings.groupby(['Owner'])[['PropertyGFABuilding_s_', 'TotalGHGEmissions__metric_tons_', 'Electricity_kBtu_', 'NaturalGas_kBtu_', 'SteamUse_kBtu_']].apply(lambda x: x.sum())
])

/var/folders/dv/sn1_d0xn0vncjbncdpkkg_pc0000gn/T/ipykernel_99492/2877209992.py:2: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  non_amazon_buildings.groupby(['Owner'])['PropertyGFABuilding_s_', 'TotalGHGEmissions__metric_tons_', 'Electricity_kBtu_', 'NaturalGas_kBtu_', 'SteamUse_kBtu_'].apply(lambda x: x.sum()),


In [50]:
amazon_v_landlords.to_csv('amazon_v_landlords.csv')